foundation/plugins/org.jboss.tools.foundation.core/src/org/jboss/tools/foundation/core/JVMProblemDetector.java

```
@@ -0,0 +1,475 @@
+	private static final String STRING_3 = "Unresolved requirement: Require-Capability: osgi.ee; filter:=\"(&(osgi.ee="; //$NON-NLS-1$
+	private static final String STRING_4 = ")(version="; //$NON-NLS-1$
+	
+	private static final long MAX_LENGTH = 1024 * 1024;
+	
+	private UnresolvedStructure structure = new UnresolvedStructure();
+	
+	private LinkedBlockingDeque<String> queue = new LinkedBlockingDeque<String>();
+	
+	private String javaVersion = null;
+	
+	@Override
+	public void earlyStartup() {
+		PlatformUI.getPreferenceStore().setValue(PREFERENCE_NAME, false);
+		if(isAllowedToShow()){
+			javaVersion = System.getProperty("java.version"); //$NON-NLS-1$
+			
+			// listen log
+			Platform.addLogListener(this);
+			
+			// read error log
+			readLogFile();
+			
+			// read queue
+			readQueue();
+		}
+	}
+	
+	private void readLogFile() {
+		File logFile = Platform.getLogFileLocation().toFile();
+		BufferedReader reader = null;
+		try{
+			reader = new BufferedReader(new InputStreamReader(new TailInputStream(logFile, MAX_LENGTH), "UTF-8")); //$NON-NLS-1$
+			while (true) {
+				String line0 = reader.readLine();
+				if (line0 == null)
+					break;
+				String line = line0.trim();
+				
+				scanLine(line);
+			}
+		} catch (FileNotFoundException e) { // do nothing
+		} catch (IOException e) { // do nothing
+		} finally {
+			try {
+				if (reader != null)
+					reader.close();
+			} catch (IOException e1) { // do nothing
+			}
+		}
+		
+		structure.report();
+	}
+	
+	private void readQueue(){
+		BufferedReader reader = null;
+		try {
+			while(isAllowedToShow()){
+				String message = queue.take();
+				reader = null;
+				try{
+					reader = new BufferedReader(new StringReader(message));
+					while (true) {
+						String line0 = reader.readLine();
+						if (line0 == null)
+							break;
+						String line = line0.trim();
+						scanLine(line);
+					}
+					structure.report();
+				} catch (IOException e) { // do nothing
```

The same as above. Empty catch and try-with-resources statement.